cmd/stc/stc.go

```diff
@@ -306,20 +306,26 @@ func signTx(net *StellarNet, key string, e *TransactionEnvelope) error {
 	return nil
 }
 
-func editor(args ...string) {
+func editor(path string, line int) {
 	ed, ok := os.LookupEnv("STCEDITOR")
 	if !ok {
 		ed, ok = os.LookupEnv("EDITOR")
 	}
 	if !ok {
 		ed = "vi"
 	}
-	if path, err := exec.LookPath(ed); err == nil {
-		ed = path
+	edArgv := strings.Split(ed, " ")
+	var argv []string
+	switch edArgv[0] {
+	case "vi", "vim":
+		argv = append(edArgv, fmt.Sprintf("+%d", line), path)
+	default:
```

Comment:
A lot of text editors take +N to edit on line number N (vi, nvi, vim, ex, emacs, emaccsclient, nano).  Others can be made to take it given the proposal to split spaces (e.g., `kate -l`, `leafpad --jump`, ).  So please blacklist vscode rather than whitelisting vi.